circle.yml

```diff
@@ -1,28 +1,20 @@
 machine:
   node:
-    version: 7.4.0
+    version: 8.1.3
   environment:
-    YARN_VERSION: 0.19.1
-    PATH: "${PATH}:${HOME}/.yarn/bin:${HOME}/${CIRCLE_PROJECT_REPONAME}/node_modules/.bin"
+    PATH: "${PATH}:${HOME}/${CIRCLE_PROJECT_REPONAME}/node_modules/.bin"
 
 dependencies:
-  pre:
-    - |
-      if [[ ! -e ~/.yarn/bin/yarn || $(yarn --version) != "${YARN_VERSION}" ]]; then
-        curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version $YARN_VERSION
-      fi
   override:
-    - yarn install
+    - yarn
   cache_directories:
-    - ~/.yarn
     - ~/.cache/yarn
 
 test:
   override:
     - yarn run lint
-    - yarn run analyze
-    - yarn run deploy:test
-    - yarn test
+    - bash -e bin/runTests.sh
```

Comment:
@samcdavid the `runTests.sh` looks like:
```
# This script is used by CircleCI to execute automated tests.

# Build all assets
yarn run build

# Boot a static file server
php -S localhost:3001 -t build/ &

# Save the PID of the server to a variable
APP_TEST_PID=$(echo $!)

# Execute tests
PORT=3001 CI=true yarn run test

# Kill the server
kill $APP_TEST_PID
```
and we're using it instead of the default test command because we're starting a simple HTTP server within Circle for each build that we can run the feature tests against (using nightmare). When the tests are done, the HTTP server is killed. This way we don't have to deploy to some external site to test against, which would cause collisions if several people are building at the same time.